### PR TITLE
DB week3 Homework

### PR DESCRIPTION
**Pull request by hwaleedali**

### CODE REVIEW

File: databases/week3/hyf_lesson3_mealsharing.sql [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `reservation` (
+  `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `number_of_guests` INT(10) NOT NULL,
+  `created_date` DATE NOT NULL,
+  `contact_phonenumber` VARCHAR(255) NOT NULL,
+  `contact_name` VARCHAR(255) NOT NULL,
+  `contact_email` VARCHAR(255) NOT NULL,
+  `meal_id` INT(10) UNSIGNED NOT NULL,
+  CONSTRAINT `fk_meal` FOREIGN KEY (`meal_id`) REFERENCES `meal` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
```
Comment (martamatos) [inline databases/week3/hyf_lesson3_mealsharing.sql]: If we delete all reservations for a given meal, the meal entry will also be deleted, is that what you want? (It is ok, just checking )

File: databases/week3/hyf_lesson3_quries.sql [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+SELECT
+  meal.title,
+  meal.price
+from
+  meal
+WHERE
+  price < 90;
+-- Get meals that still has available reservations
+  
+-- SELECT
```
Comment (martamatos) [inline databases/week3/hyf_lesson3_quries.sql]: This is the trap query. You need to sum the number_of_guests over all reservations for a given meal and then compare that number with the max_reservations of that meal.

There are at least two ways of doing this, both involve stuff that we didnt talk about in class:
 - use the COALESCE() function: https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_coalesce
 - use GROUP BY, SUM, and `HAVING: https://www.w3schools.com/sql/sql_having.asp

Also remember that there might be meals that don't have any reservations yet.

File: databases/week3/hyf_lesson3_quries.sql [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+  meal
+WHERE
+  created_date > '2022-02-14' && created_date < '2022-02-16';
+-- Get only specific number of meals fx return only 5 meals
+SELECT
+  meal.title
+from
+  meal
+where
+  price < 90
```
Comment (martamatos) [inline databases/week3/hyf_lesson3_quries.sql]: The exercise doesn't say anything about the price.

File: databases/week3/hyf_lesson3_quries.sql [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+ORDER BY
```
Comment (martamatos) [inline databases/week3/hyf_lesson3_quries.sql]: The exercise doesn't say anything about ordering the results.

File: databases/week3/hyf_lesson3_quries.sql [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+  reservation.created_date DESC;
+-- Sort all meals by average number of stars in the reviews
+SELECT
+  meal.title,
+  round(AVG(review.stars), 1)
+from
+  meal
+  join review on meal.id = review.meal_id
+group BY
+  meal.title;
```
Comment (martamatos) [inline databases/week3/hyf_lesson3_quries.sql]: Don't forget to order by the average number of stars.

**Review by martamatos — CHANGES_REQUESTED** [review]
Good job overall Hafiz!
I've left only a few comments :)

File: databases/week3/hyf_lesson3_mealsharing.sql (hunk 1 above)
Comment (hwaleedali) [reply databases/week3/hyf_lesson3_mealsharing.sql]: i fix the problem

File: databases/week3/hyf_lesson3_quries.sql (hunk 2 above)
Comment (hwaleedali) [reply databases/week3/hyf_lesson3_quries.sql]: i fix the problem

File: databases/week3/hyf_lesson3_quries.sql (hunk 3 above)
Comment (hwaleedali) [reply databases/week3/hyf_lesson3_quries.sql]: i fix the problem

File: databases/week3/hyf_lesson3_quries.sql (hunk 4 above)
Comment (hwaleedali) [reply databases/week3/hyf_lesson3_quries.sql]: i fix the problem

File: databases/week3/hyf_lesson3_quries.sql (hunk 5 above)
Comment (hwaleedali) [reply databases/week3/hyf_lesson3_quries.sql]: i fix the problem

**Pull request merged**